File: src/AgentMulder.ReSharper.Tests/AuotfacVB/ContainerBuilderTests.cs
using System.Linq;
using AgentMulder.Containers.Autofac;
using AgentMulder.ReSharper.Domain.Containers;
using AgentMulder.ReSharper.Tests.AuotfacVB.Helpers;
using JetBrains.ReSharper.Psi;
using JetBrains.ReSharper.Psi.Tree;
using JetBrains.ReSharper.Psi.VB.Tree;
using NUnit.Framework;

namespace AgentMulder.ReSharper.Tests.AuotfacVB
{
    [TestAutofacVB]
    public class ContainerBuilderTests : ComponentRegistrationsTestBaseVB
    {
        protected override string RelativeTestDataPath
        {
            get { return @"VB\Autofac"; }
        }

        protected override IContainerInfo ContainerInfo
        {
            get { return new AutofacContainerInfo(); }
        }

        [TestCase("RegisterTypeGeneric", new[] { "CommonImpl1.vb" })]
        [TestCase("RegisterTypeNonGeneric", new[] { "CommonImpl1.vb" })]
        [TestCase("RegisterWithLambda", new[] { "Foo.vb" })]
        [TestCase("RegisterWithLambdaTakesDependency", new[] { "TakesDependency.vb" })]
        [TestCase("RegisterWithLambdaInitializer", new[] { "FooBar.vb" })]
        [TestCase("RegisterComplex", new[] { "GoldCard.cs", "StandardCard.vb" })]
        [TestCase("RegisterComplexWithVariable", new[] { "GoldCard.vb" })]
        public void DoTest(string testName, string[] fileNames)
        {
            RunTest(testName, registrations =>
            {
                IVBFile[] codeFiles = fileNames.Select(GetCodeFile).ToArray();

                Assert.AreEqual(codeFiles.Length, registrations.Count());
                foreach (var codeFile in codeFiles)
                {
                    codeFile.ProcessChildren<ITypeDeclaration>(declaration =>
                        Assert.That(registrations.Any((r => r.Registration.IsSatisfiedBy(declaration.DeclaredElement)))));
                }
            });
        }
    }
}

File: src/AgentMulder.Resharper.VB.Tests/Autofac/ContainerBuilderTests.vb
Imports AgentMulder.Resharper.Domain.Containers
Imports AgentMulder.Containers.Autofac
Imports NUnit.Framework
Imports JetBrains.ReSharper.Psi.Tree
Imports JetBrains.ReSharper.Psi.VB.Tree
Imports JetBrains.ReSharper.Psi

Namespace Autofac
    Public Class ContainerBuilderTests
        Inherits ComponentRegistrationsTestBase

        Protected Overrides ReadOnly Property RelativeTestDataPath As String
            Get
                Return "Autofac"
            End Get
        End Property

        Protected Overrides ReadOnly Property ContainerInfo As IContainerInfo
            Get
                Return New AutofacContainerInfo()
            End Get
        End Property

        <TestCase("RegisterTypeGeneric", New String() {"CommonImpl1.vb"})>
        Public Sub DoTest1(testName As String, fileNames As String())
            RunTest(testName, Function(registrations)
                                  Dim codefiles() As IVBFile = fileNames.Select(Function(x) GetCodeFile(x)).ToArray()
                                  Assert.AreEqual(codefiles.Length, registrations.Count())
                                  For Each codeFile In codefiles
                                      codeFile.ProcessChildren(Of ITypeDeclaration)(Function(declaration)
                                                                                        Assert.That(registrations.Any(Function(r) r.Registration.IsSatisfiedBy(declaration.DeclaredElement)))
                                                                                    End Function)
                                  Next
                              End Function)
        End Sub

    End Class
End Namespace